src/cryptonote_core/blockchain.cpp

```diff
@@ -4450,7 +4547,10 @@ bool Blockchain::update_next_cumulative_weight_limit(uint64_t *long_term_effecti
       long_term_median = get_long_term_block_weight_median(db_height - nblocks - 1, nblocks);
     }
 
+    const uint64_t previous_long_term_effective_median_block_weight = m_long_term_effective_median_block_weight;
     m_long_term_effective_median_block_weight = std::max<uint64_t>(CRYPTONOTE_BLOCK_GRANTED_FULL_REWARD_ZONE_V5, long_term_median);
+    if (hf_version >= HF_VERSION_2021_SCALING)
+      m_long_term_effective_median_block_weight = std::max<uint64_t>(m_long_term_effective_median_block_weight, previous_long_term_effective_median_block_weight / 2);
 
```

Comment:
I don't think these changes belong here. As noted in another comment, it should be in `get_next_long_term_block_weight()`.

Reply:
At first glance, a get_next_... function should not modify a member var. Is it actually incorrect if it's here ?

Reply:
I did not recommend modifying a member var in `get_next_long_term_block_weight()`. I recommended moving this functionality into `get_next_long_term_block_weight()`, which is used to set `long_term_median` in line 4450 referenced by this comment, which is used to set `m_long_term_effective_median_block_weight` in line 4453 referenced by this comment.

Reply:
If you mean moving the entire calc there, IIRC it's called a lot and would noticeably slow down, so I'm leaving it here.

Reply:
Sorry, what? A call to `std::max<uint64_t>()` will "noticeably slow [it] down"?

Reply:
```suggestion

    m_long_term_effective_median_block_weight = std::max<uint64_t>(CRYPTONOTE_BLOCK_GRANTED_FULL_REWARD_ZONE_V5, long_term_median);

```

Reply:
After your update to `get_next_long_term_block_weight()` these lines don't do anything any more.